src/main/java/me/confor/velocity/chat/modules/GlobalChat.java

```diff
@@ -76,14 +77,20 @@ public void onPlayerChat(PlayerChatEvent event) {
     }
 
     @Subscribe
-    public void onConnect(LoginEvent event) {
+    public void onConnect(PlayerChooseInitialServerEvent event) {
```

Comment:
I think it makes more sense to use [`ServerConnectedEvent`](https://jd.papermc.io/velocity/3.0.0/com/velocitypowered/api/event/player/ServerConnectedEvent.html) for this listener since we don't have to change what server the player is connecting to (I believe that's the intent of the `PlayerChooseInitialServerEvent`)

Reply:
I was under the assumption `ServerConnectedEvent` was to be used when a player is *switching* to a server, rather than first joining a server via the proxy, given the existence of the `getPreviousServer()` method.
But what you're saying about `PlayerChooseInitialServerEvent` existing to change the connection's destination makes much more sense.
(I will admit that I missed quite a few obvious things when choosing which event to listen to.)

In any case, if we *are* to use `ServerConnectedEvent` instead, would it make sense to merge the `onConnect()` and `onServerConnect()` listeners? We'd only need to build off of the `if (currentServer.isEmpty())` statement, no?

Reply:
> I was under the assumption ServerConnectedEvent was to be used when a player is switching to a server, rather than first joining a server via the proxy, given the existence of the getPreviousServer() method.

`getPreviousServer()` returns an `Optional<RegisteredServer>`, so if I'm understanding correctly it should return Empty if there was no previous server.